src/modules/SdnDiag.NetworkController.psm1

```diff
@@ -2746,35 +2753,57 @@ function Set-SdnResource {
+
+        # if we doing an update or delete operation
+        # ensure that the resource exists prior to performing operation
+        if ($OperationType -eq 'Update' -or $OperationType -eq 'Delete') {
+            try {
+                $null = Invoke-RestMethodWithRetry @restParams
             }
-            else {
+            catch [System.Net.WebException] {
+                if ($_.Exception.Response.StatusCode -eq "NotFound") {
+                    throw New-Object System.NotSupportedException("Resource was not found. Ensure the resource exists before attempting to perform $($OperationType) operation.")
+                }
+                else {
+                    throw $_
+                }
+            }
+            catch {
                 throw $_
             }
         }
-        catch {
-            throw $_
-        }
 
-        $modifiedObject = Remove-PropertiesFromObject -Object $Object -PropertiesToRemove @('ConfigurationState','ProvisioningState')
-        $jsonBody = $modifiedObject | ConvertTo-Json -Depth 100
-        $putRestParams.Body = $jsonBody
-
-        if ($PSCmdlet.ShouldProcess($uri, "Invoke-RestMethod will be called with PUT to configure the properties of $($putRestParams.Uri)`n`t$jsonBody")) {
-            $null = Invoke-RestMethodWithRetry @putRestParams
-            if (Confirm-ProvisioningStateSucceeded -NcUri $putRestParams.Uri @confirmParams) {
-                $result = Invoke-RestMethodWithRetry @restParams
-                return $result
+        switch ($OperationType) {
+            'Delete' {
+                $removeRestParams = $restParams.Clone()
+                $removeRestParams.Method = 'DELETE'
+                if ($PSCmdlet.ShouldProcess($uri, "Invoke-RestMethod will be called with $($removeRestParams.Method) for the resource at $($removeRestParams.Uri)")) {
+                    $null = Invoke-RestMethodWithRetry @removeRestParams
+                }
+            }
+            default {
+                $modifiedObject = Remove-PropertiesFromObject -Object $Object -PropertiesToRemove @('ConfigurationState','ProvisioningState')
+                $jsonBody = $modifiedObject | ConvertTo-Json -Depth 100
+
+                $putRestParams = $restParams.Clone()
+                $putRestParams.Method = 'PUT'
+                $putRestParams.Add("Headers", @{"Content-Type"="application/json"})
+                $putRestParams.Add("ContentType", "application/json; charset=UTF-8")
+                $putRestParams.Add("Body", $jsonBody)
+
+                if ($PSCmdlet.ShouldProcess($uri, "Invoke-RestMethod will be called with $($putRestParams.Method) for the resource at $($putRestParams.Uri)`n`t$jsonBody")) {
+                    $null = Invoke-RestMethodWithRetry @putRestParams
+                    if (Confirm-ProvisioningStateSucceeded -NcUri $putRestParams.Uri @confirmParams) {
```

Comment:
If I understand correctly, Confirm-ProvisioningStateSucceeded does a GET, checks the provisioning state, then we do a GET again with Invoke-RestMethodWithRetry to return the resource. Can we simplify this logic to just use the request result from Confirm-ProvisioningStateSucceeded so we don't need to do an extra GET? Also, should this cmdlet always wait for the provisioning state to succeed before proceeding? I believe the majority of NC operations are designed to be asynchronous.